src/lib/y2storage/device.rb

```diff
@@ -170,19 +174,15 @@ def siblings
       storage_siblings(itself)
     end
 
-    # @!method detect_resize_info
-    #   Information about the possibility of resizing a given device. Returns
-    #   nil if the device does not exist in the probed devicegraph.
-    #   @see ResizeInfo
-    #   @see exists_in_probed?
-    #   @note This is slightly different from Storage::detect_resize_info, which
-    #     requires to be called in a device that belongs to the probed devicegraph.
+    # Check if the device can be resized.
     #
-    #   @return [ResizeInfo]
-    def detect_resize_info
-      return nil unless exists_in_probed?
-      probed_device = StorageManager.instance.probed.find_device(sid)
-      probed_device.storage_detect_resize_info
+    # If the device has any children, they are also taken into account;
+    # the result of this method is the combined information about this device
+    # and all its children.
+    #
+    # @return [Boolean] true if the device can be resized, false if not.
+    def can_resize?
```

Comment:
Just an afterthought on this. If `detect_resize_info` is an expensive operation (I don't know, It's just what the name seems to suggest), maybe "hiding" the call on this method is not such a good idea. Just looking at the resulting API I would be tempted to call `#can_resize?` over and over instead of operating with the cached `ResizeInfo` object.

Reply:
I'll add a note to the documentation.